check/src/unify_type.rs

```diff
@@ -370,51 +342,6 @@ fn unify_app_<'a, 's, U>(unifier: &mut UnifierState<'a, 's, U>,
     }
 }
 
-fn zip_function<'a, 's, U>(unifier: &mut UnifierState<'a, 's, U>,
-                           arg: &TcType,
-                           ret: &TcType,
-                           other: &TcType,
-                           matcher: &mut FnMut(&mut UnifierState<'a, 's, U>, &TcType, &TcType)
-                                               -> Option<TcType>)
-                           -> Result<Option<TcType>, Error<Symbol>>
-    where U: Unifier<&'a (TypeEnv + 'a), TcType>
-{
-    let error = || {
-        let func = Type::function(vec![arg.clone()], ret.clone());
-        debug!("Unify error: {} <=> {}", func, other);
-        Err(UnifyError::TypeMismatch(func, other.clone()))
-    };
-    let subs = unifier.subs;
-    let other = subs.real(other);
-    let (other_arg, fn_prim, other_ret) = match **other {
-        Type::App(ref fn_prim, ref args) if args.len() == 2 => (Some(&args[0]), fn_prim, &args[1]),
-        Type::App(ref fn_prim, ref args) if args.len() == 1 => (None, fn_prim, &args[0]),
-        _ => return error(),
-    };
-    match other_arg {
-        Some(other_arg) => {
-            matcher(unifier,
-                    fn_prim,
-                    &Type::builtin(types::BuiltinType::Function));
-            let new_arg = matcher(unifier, arg, other_arg);
-            let new_ret = matcher(unifier, ret, other_ret);
-            Ok(merge(arg,
-                     new_arg,
-                     ret,
-                     new_ret,
-                     |args, ret| Type::function(vec![args], ret)))
-        }
-        None => {
-            let new_fn = matcher(unifier,
-                                 fn_prim,
-                                 &Type::app(Type::builtin(types::BuiltinType::Function),
-                                             vec![arg.clone()]));
-            let new_ret = matcher(unifier, ret, other_ret);
-            Ok(merge(fn_prim, new_fn, ret, new_ret, |l, r| Type::app(l, vec![r])))
-        }
-    }
-}
```

Comment:
Its fine to remove, it just existed to handle the special case of `Function <=> App` unification.